framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -4631,6 +4619,452 @@ private void makeGroundTargetType(
+
+    captureTypeArgCopier.copy(uncapturedType, capturedType);
+
+    AnnotatedDeclaredType typeDeclaration =
+        (AnnotatedDeclaredType) getAnnotatedType(uncapturedType.getUnderlyingType().asElement());
+
+    // A mapping from type variable to its type argument in the captured type.
+    Map<TypeVariable, AnnotatedTypeMirror> typeVariableMapping = new HashMap<>();
+    // A mapping from a captured type variable to the annotated captured type variable.
+    Map<TypeVariable, AnnotatedTypeVariable> capturedTypeVarMapping = new HashMap<>();
+    // `newTypeArgs` will be the type arguments of the result of this method.
+    List<AnnotatedTypeMirror> newTypeArgs = new ArrayList<>();
+    for (int i = 0; i < typeDeclaration.getTypeArguments().size(); i++) {
+      TypeVariable typeVarTypeMirror =
+          (TypeVariable) typeDeclaration.getTypeArguments().get(i).getUnderlyingType();
+      AnnotatedTypeMirror uncapturedTypeArg = uncapturedType.getTypeArguments().get(i);
+      AnnotatedTypeMirror capturedTypeArg = capturedType.getTypeArguments().get(i);
+      if (uncapturedTypeArg.getKind() == TypeKind.WILDCARD) {
+        if (TypesUtils.isCapturedTypeVariable(capturedTypeArg.getUnderlyingType())) {
+          // The type argument is a captured type variable. Use the type argument from the newly
+          // created and yet-to-be annotated capturedType. (The annotations are added as
+          // part of capturing the wildcard.)
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          // Also, add a mapping from the captured type variable to the annotated captured
+          // type variable, so that if one captured type variable refers to another, the same
+          // AnnotatedTypeVariable object is used.
+          capturedTypeVarMapping.put(
+              ((AnnotatedTypeVariable) capturedTypeArg).getUnderlyingType(),
+              (AnnotatedTypeVariable) capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+        } else {
+          // Javac used a declared type instead of a captured type variable.  This seems to happen
+          // when the bounds of the captured type variable would have been identical. This seems to
+          // be a violation of the JLS, but javac does this, so the Checker Framework handle that
+          // case.
+          typeVariableMapping.put(typeVarTypeMirror, capturedTypeArg);
+          newTypeArgs.add(capturedTypeArg);
+          replaceAnnotations(
+              ((AnnotatedWildcardType) uncapturedTypeArg).getSuperBound(), capturedTypeArg);
+        }
+      } else {
+        // The type argument is not a wildcard.
+        typeVariableMapping.put(typeVarTypeMirror, uncapturedTypeArg);
+        if (uncapturedTypeArg.getKind() == TypeKind.TYPEVAR) {
+          AnnotatedTypeVariable typeVar = (AnnotatedTypeVariable) uncapturedTypeArg;
+          typeVariableMapping.put(typeVar.getUnderlyingType(), typeVar);
```

Comment:
What is the relationship between `typeVar.getUnderlyingType()` and `typeVarTypeMirror`?  I'm unclear whether this is replacing a mapping or is adding a new mapping.

Reply:
If `uncapturedType.getUnderlingType()` is `G<?, X>` , and  the declaration of `G` is `G<T, V>`, then `typeVar.getUnderlingType()` is `X` and `typeVarTypeMirror` is `V`.   

`typeVar.getUnderlyingType()` and `typeVarTypeMirror` can be the same.  For example if `G` is declared as `G<T, V extends G<?, V>>`, then `uncapturedType` could be `G<?, V>`, in which case both `typeVar.getUnderlyingType()` and `typeVarTypeMirror` are `V`.

Reply:
Thanks for the clarification.  I would be inclined to add a comment given that I had a question -- unless you feel that would be unhelpful clutter.